src/main/java/com/bina/varsim/tools/evaluation/SAMcompare.java

```diff
@@ -358,5 +367,45 @@ public void setBedFilename(String bedFilename) {
         }
     }
 
+    /**
+     * check if a read is supposed to be aligned in the user-supplied BED region
+     * @param intersector BED-file object
+     * @param trueLoci collection of true alignment loci
+     * @param rec one alignment in SAM
+     * @return
+     */
+    private static boolean isContainedInBed(BedFile intersector, Collection<GenomeLocation> trueLoci, SAMRecord rec) {
+        if (intersector == null) {
+            return true;
+        }
+        boolean isContainedInBed = false;
+        for (GenomeLocation loc : trueLoci) {
+            if (intersector.contains(loc.chromosome, loc.location
+                    //here we should have used the true alignment length
+                    //however that is difficult to know unless we have a perfect BAM
+                    , loc.location + rec.getReadLength() - 1)) {
+                isContainedInBed = true;
+            }
+        }
+        return isContainedInBed;
+    }
 
+    /**
+     * calculate identity ratio of the read
+     * formula:
+     *
+     * sum of sequence matches (= or M in CIGAR) / total length of the read
+     *
+     * @param alignment
+     * @return
+     */
+    private static double getIdentity(SAMRecord alignment) {
+        int nMatches = 0;
+        for (CigarElement c : alignment.getCigar().getCigarElements()) {
+            if (ALL_POSSIBLE_MATCHES.contains(c.getOperator())) {
+                nMatches += c.getLength();
+            }
+        }
+        return (double) nMatches/alignment.getReadLength();
```

Comment:
That's a good point! Here I am trying to be strict about which read is truly multi-aligned because we only calculate a ratio and totally ignore # of hits, base qualities, read lengths, etc. Sometimes a false multi-read might be found if we ignore soft-clipped bp. For example, 
```
ref1: TTTATCGTA
read: aaaATCGTA   #lower case for softclipping
ratio: 100%

ref2: AAAATC-TA
read: AAAATCGTA
ratio: 88%
```

the 2nd alignment is likely to have higher MAPQ and is perhaps represents the truth alignment. but the 1st alignment actually has higher identity ratio and might be considered as a true multi-read if we ignore softclipped bp in the denominator. Is this analysis reasonable?